### PR TITLE
browser-support: remove lookbehind regex statement

### PR DESCRIPTION
**Pull request by 0xdeafcafe**
Small change to bring in support for browsers. The current code for converting camelCase to snake_case uses a regex which includes a positive lookbehind, which is not supported in all browsers

Screenshot 2021-09-22 at 10 22 06

This change add's a fork to the logic if it detects a browser environment, and uses a two-pass regex approach to achieve the same goal. Both sides of the fork are run against the same test suit to ensure consistency.

### CODE REVIEW

**Review by mfiedorowicz — CHANGES_REQUESTED** [review]
Removed tests? It's gonna blow Wakam LTM quotes...

**Comment by 0xdeafcafe** [comment]
> Removed tests? It's gonna blow Wakam LTM quotes...

😅. The only tests were for code that no longer exists

**Comment by mfiedorowicz** [comment]
> > Removed tests? It's gonna blow Wakam LTM quotes...
> 
> 😅. The only tests were for code that no longer exists

Then it's even worse as we depend on that code.

**Comment by 0xdeafcafe** [comment]
> > > Removed tests? It's gonna blow Wakam LTM quotes...
> > 
> > 
> > 😅. The only tests were for code that no longer exists
> 
> Then it's even worse as we depend on that code.

I'm not really sure we should be adding tests to test external packages, which have tests in their respective repos.

**Comment by mfiedorowicz** [comment]
Can we ensure behaviour of these new external packages work correctly for cases we had to solve internally in https://github.com/cuvva/crpc/pull/4?

**Comment by 0xdeafcafe** [comment]
Good point. I checked and the `snakecase_keys` package does seem to have the issue of incorrectly parsing the `distanceToLondonIs100km` key. Am happy to revert to the regex approach, but that had two issues;

- It incorrectly parses `clientDataJSON` to `client_data_j_s_o_n`
- Uses a positive lookbehind assertion, which isn't supported in all browsers (IE and Safari don't)

Using the library seems like it's not going to be feasible, due to the issue you flagged, so I'm going to revert and see if I can get the regex to work without the lookbehind + I'll add tests to cover the case I flagged above.

**Comment by tgmendes** [comment]
Just want to chime in and say that I think it would be very important to have a short description in this PR as to why we're doing this. Just to have a package instead of our own lib? Or did something go wrong? (note that both are valid reasons). It's important for both reviewers and for a good historical log.

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1 @@
+test
```
Comment (nchagrass) [inline .npmignore:1]: 👀

Comment (0xdeafcafe) [reply .npmignore:1]: We shouldn't bundle the tests in the NPM package

**Review by nchagrass — APPROVED** [review]
Tests have been conserved, nice work! 👍  (I think Michal is off)

**Pull request merged**